### PR TITLE
Add example GitHub workflow for automated update PRs

### PR DESCRIPTION
**Pull request by sd109**
Can be used in downstream azimuth-config repos for auto-creation of update PRs when new releases are published upstream. For example PR created by this workflow, see https://github.com/stackhpc/leafcloud-stackhpc-azimuth-config/pull/32

### CODE REVIEW

**Comment by sd109** [comment]
Marking as draft until I get round to adding some docs

**Comment by sd109** [comment]
VSCode format-on-save seems to have made some unrelated changes to the markdown docs, I can remove those changes if preferred.

File: docs/deployment/automation.md [hunk 1]
```diff
@@ -6,14 +6,14 @@ deployment rather than running deployment commands manually. Using this approach
 configuration changes are automatically deployed to test, staging and production
 environments, although deployments to production typically include a manual approval.
 
-!!! note  "Continuous delivery vs continuous deployment"
+!!! note "Continuous delivery vs continuous deployment"
```
Comment (mkjpryor) [inline docs/deployment/automation.md:9]: I think the double space here is actually important to the rendering

File: docs/deployment/automation.md [hunk 2]
```diff
@@ -6,14 +6,14 @@ deployment rather than running deployment commands manually. Using this approach
...
 
     Continuous delivery is very similar to
     [Continuous Deployment](https://en.wikipedia.org/wiki/Continuous_deployment) with
     the exception that in continuous deployment, production deployments are also fully
     automated with no manual intervention or approval.
 
-!!! tip  "Using a site mixin"
+!!! tip "Using a site mixin"
```
Comment (mkjpryor) [inline docs/deployment/automation.md:16]: Same here

**Review by mkjpryor — DISMISSED** [review]

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**